### DISCUSSION TITLE
Welcome to UrlTracker Discussions!

### DISCUSSION THREAD

**Discussion by D-Inventor** [Infocaster/UrlTracker#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by D-Inventor** [comment]
I'm D_Inventor, I am an Umbraco enthusiast and am generally passionate about anything related to .NET.

I am the developer behind the upcoming 5.0.0 and 9.0.0 releases of the URL Tracker.
Though the URL Tracker has been an amazing package so far, I saw the signs of age in our code base and the struggle to keep it up-to-date with the later Umbraco releases. The URL Tracker was celebrated in Umbraco 7 as one of the top 10 packages that you need to have in every installation. It is my aim to revive this status for the URL Tracker for Umbraco 8, 9 and beyond by rewriting the code base to something performant, maintainable, customizable and extendible.

I love interaction from the community, so I hope that our URL Tracker earns back its place in the spotlight and that we may get many questions, suggestions, criticism and compliments!

**Comment by MarcoV-sepa** [comment]
Would it be possible for you to put up an example of how to limit who can see the dashboard in Umbraco 9/10 please

**Reply by D-Inventor** [reply]
Hi @MarcoV-sepa ,

That's a cool idea, not something I've thought about before.

It's not supported out-of-the-box, but you should be able to leverage Umbraco's DI container to get the desired result. It comes down to removing the Dashboard from the Dashboards collection and replacing it with your own implementation.

Let me do some research and come back to you when I have some pointers.

**Reply by D-Inventor** [reply]
Ok, so here's what I've found:

To create custom access rules, you have to replace our dashboard with a custom implementation. The properties cannot be overridden, so you'll have to copy [the dashboard source](https://github.com/Infocaster/UrlTracker/blob/v10/develop/src/UrlTracker.Web/Package/UrlTrackerDashboard.cs) and then make your edits. You can [check out how to implement access rules in a dashboard in the official Umbraco documentation](https://our.umbraco.com/documentation/extending/dashboards/#specifying-permissions).

Next step is to remove the original dashboard. You can [check out how to remove existing dashboards in the official Umbraco documentation](https://our.umbraco.com/documentation/extending/dashboards/#remove-an-umbraco-dashboard). Make sure your composer runs after the composer from the URL Tracker using an attribute:

```csharp
[ComposeAfter(typeof(UrlTracker.Web.WebComposer))]
public class MyComposer : IComposer
...
```

Now you just add your own dashboard and you're done.

I hope that helps you. Let me know if there's more I can do for you!

**Comment by Shadyn** [comment]
I have an old version of URL Tracker running on a V7 site.  I now have a V10 site with the new URL Tracker installed.  I made a few demo redirects and exported to csv so I could see the format.  I made a csv of my old urls and formatted to be the same (semicolons as seperators, different fields and names, etc).  The import seems to throw an error saying a method isn't implemented.

Method not found: 'Void CsvHelper.CsvReader..ctor(System.IO.TextReader, CsvHelper.Configuration.CsvConfiguration)'.
Exception Details
System.MissingMethodException, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e: Method not found: 'Void CsvHelper.CsvReader..ctor(System.IO.TextReader, CsvHelper.Configuration.CsvConfiguration)'. 

Any ideas on how to get around this?

**Reply by D-Inventor** [reply]
Hi @Shadyn ! Thank you for reaching out. Could you please [create a new issue](https://github.com/Infocaster/UrlTracker/issues/new/choose) (The bug report template would do) and specify which version of Umbraco and which version of the URL Tracker you use and which steps to take to reproduce your issue? That would be very helpful for me to diagnose your issue.
